src/test/java/com/ihren/task2/exercise2/TransactionConverterTest.java

```diff
@@ -0,0 +1,83 @@
+package com.ihren.task2.exercise2;
+
+import com.ihren.task2.exercise2.exception.BusinessException;
+import com.ihren.task2.exercise2.mapper.TransactionMapper;
+import com.ihren.task2.exercise2.model.Element;
+import com.ihren.task2.exercise2.model.Item;
+import com.ihren.task2.exercise2.model.Transaction;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.mock;
+
+@ExtendWith(MockitoExtension.class)
+class TransactionConverterTest {
+
+    @InjectMocks
+    private TransactionConverter transactionConverter;
+
+    @Mock
+    private TransactionMapper transactionMapper;
+
+    @Mock
+    private Item posLogRoot;
+
+    private static Stream<Arguments> dataProvider() {
+        return Stream.of(
+                Arguments.of(mock(Transaction.class)),
+                Arguments.of((Object) null)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("dataProvider")
+    void should_ReturnTransaction_when_InputIsValid(Transaction expected) {
+        //given
+        Item itemMock = mock(Item.class);
+        Element elementMock = mock(Element.class);
+        Map<String, Object> headersMock = mock(Map.class);
+
+        given(itemMock.element()).willReturn(elementMock);
+        given(transactionMapper.map(elementMock, headersMock)).willReturn(expected);
+
+        //when
+        Transaction actual = transactionConverter.convert(itemMock, headersMock);
+
+        //then
+        assertEquals(expected, actual);
+        then(itemMock).should().element();
```

Comment:
it is not necessary to verify it, you already mocked it. It will fail if it is not mocked.